### PR TITLE
Bump to ADAM 0.23.0 release.

### PR DESCRIPTION
**Pull request by fnothaft**
CC @droazen RE https://github.com/bigdatagenomics/adam/issues/1762

### CODE REVIEW

**Comment by lbergelson** [comment]
@fnothaft Thanks for this pr.  I fixed the problem that was causing compilation to fail, but now we're getting real errors.

ex:
```
    org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1419.0 failed 1 times, most recent failure: Lost task 0.0 in stage 1419.0 (TID 3897, localhost): java.lang.IllegalArgumentException: requirement failed: Failed when trying to create region 21 10006438 10006545 on null strand.
```

**Comment by fnothaft** [comment]
Thanks for the heads up @lbergelson! I'm looking at this now; we'd added stricter validation of the strand argument in the `ReferenceRegion` constructor recently, so this should be a fairly straightforward fix.

**Comment by fnothaft** [comment]
Hi @lbergelson! I've just pushed an update to this branch that appears to resolve the test failures on my side.

**Comment by lbergelson** [comment]
@fnothaft Thanks, that's great.  I'll merge this when it passes.

**Comment by fnothaft** [comment]
@lbergelson any insight as to why `org.broadinstitute.hellbender.utils.mcmc.PosteriorSummaryUtilsUnitTest` is failing? I haven't touched anything surrounding said code. Is this a flagging unit test?

**Comment by lbergelson** [comment]
I didn't think that test was flakey, but I don't really know anything about it.  I restarted to be sure.  I'm not sure what's happening exactly the output wasn't helpful...  We'll see what happens when it reruns.  Hopefully it's just some transient thing.

**Comment by fnothaft** [comment]
Sounds good! I'll be watching here.

**Comment by fnothaft** [comment]
@lbergelson I'm looking into this some more; I wonder if the ADAM bump is bringing in some dependency conflict. I'll report back.

**Comment by fnothaft** [comment]
OK! There was a transient dependency that bumped the `commons-math3` version. I've just pushed an update that should fix this.

File: build.gradle [hunk 1]
```diff
@@ -197,13 +197,14 @@ dependencies {
         exclude module: 'servlet-api'
     }
 
-    compile 'org.bdgenomics.bdg-formats:bdg-formats:0.5.0'
-    compile('org.bdgenomics.adam:adam-core-spark2_2.11:0.20.0') {
+    compile('org.bdgenomics.adam:adam-core-spark2_2.11:0.23.0') {
+        exclude group: 'org.slf4j'
         exclude group: 'org.slf4j'
```
Comment (droazen) [inline build.gradle]: This exclusion is duplicated twice.

File: src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReferenceTwoBitSource.java [hunk 2]
```diff
@@ -75,15 +75,18 @@ private static ReferenceRegion simpleIntervalToReferenceRegion(SimpleInterval in
...
         // to truncate our interval at the contig end if necessary.
-        final TwoBitRecord contigRecord = twoBitSeqEntries.get(interval.getContig());
-        Utils.nonNull(contigRecord, () -> "Contig " + interval.getContig() + " not found in reference dictionary");
-        return new SimpleInterval(interval.getContig(), interval.getStart(), Math.min(interval.getEnd(), contigRecord.dnaSize()));
+        final Option<SequenceRecord> optContigRecord = twoBitSeqDictionary.apply(interval.getContig());
+	if (optContigRecord.isEmpty()) {
+	    throw new IllegalArgumentException("Contig " + interval.getContig() + " not found in reference dictionary");
+	}
+	final SequenceRecord contigRecord = optContigRecord.get();
+        return new SimpleInterval(interval.getContig(), interval.getStart(), Math.min(interval.getEnd(), Math.toIntExact(contigRecord.length())));
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReferenceTwoBitSource.java]: Indentation a little funky here.

**Comment by droazen** [comment]
Very minor comments, otherwise looks good. Thanks for the patch!

**Comment by fnothaft** [comment]
Just pushed two commits to address your feedback, @droazen!

**Pull request merged**